### PR TITLE
validation + UI update

### PR DESCRIPTION
**Pull request by avan989**
1) Registration Validation
     a) first and last name cannot have numbers and special character
     b) password must be 10 char, 1 upper, 1 lower, 1 number, 1 special
2) App bar update
3) message UI update 
     a) Remove message # and course #.

### CODE REVIEW

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 1]
```diff
@@ -227,7 +243,32 @@ private boolean validateRegistrationForm() {
       mPasswordField.setError("Required.");
       mPasswordField.requestFocus();
       valid = false;
-    } else {
+    } else if (password.length() < 10) {
+      mPasswordField.setError("Password needs to be at least 10 char. ");
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:247]: Remove extra space at end of sentence.
Change to "Password must contain at least 10 characters."

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 2]
```diff
@@ -227,7 +243,32 @@ private boolean validateRegistrationForm() {
...
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if (!containNumber(password)){
+      mPasswordField.setError("Password need to contain 1 number");
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:252]: Replace with "...must contain at least 1 number."

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 3]
```diff
@@ -227,7 +243,32 @@ private boolean validateRegistrationForm() {
...
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if (!containLowerCase(password)){
+      mPasswordField.setError("Password need to contain 1 lower case");
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:257]: Change the rest to "..must contain" style

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 4]
```diff
@@ -227,7 +243,32 @@ private boolean validateRegistrationForm() {
...
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if (!containUpperCase(password)){
+      mPasswordField.setError("Password need to contain 1 Upper case");
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if(!containSpecialChar(password)){
+      mPasswordField.setError("Password need to contain 1 special characters (!@#$%^&*())");
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:267]: Could we have these requirements (all of these for the PW in teeny tiny text under the pw, so they don't have to keep trying and failing?)

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 5]
```diff
@@ -247,6 +288,65 @@ private boolean validateRegistrationForm() {
...
 
+  private boolean containLowerCase(String string){
+    boolean valid = false;
+    char ch;
+
+    for(int i=0; i < string.length(); i++) {
+      ch = string.charAt(i);
+
+      if(Character.isLowerCase(ch)) {
+        valid = true;
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:299]: I think we can break here, but we don't have to finish the loop. And same for the rest of the loops

File: T2CC_APP/t2cc/app/src/main/res/layout/appbar.xml [hunk 6]
```diff
@@ -14,7 +14,7 @@
         android:layout_marginLeft="5dp"
         android:layout_width="48dp"
         android:layout_height="50dp"
-        android:layout_marginRight="215dp"
+        android:layout_marginRight="200dp"
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/res/layout/appbar.xml:17]: Can we give these a bit more room. I want the messages farther than the log out. It's the main function of our app and I think right now it's ways too close to the sign out button. So maybe we need to switch the initials and sign out fields. That way a slip of a finger when trying to get your messages wont log you out

**Comment by avan989** [comment]
updated.

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 7]
```diff
@@ -227,7 +243,32 @@ private boolean validateRegistrationForm() {
...
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if (!containUpperCase(password)){
+      mPasswordField.setError("Password need to contain 1 Upper case");
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if(!containSpecialChar(password)){
+      mPasswordField.setError("Password need to contain 1 special characters (!@#$%^&*())");
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:267]: needs "must contain"...and you have different cases for upper and lower case. We should pick one and be consistent

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 8]
```diff
@@ -209,6 +217,14 @@ private boolean validateRegistrationForm() {
       mLastNameField.setError("Required.");
       mLastNameField.requestFocus();
       valid = false;
+    } else if (containNumber(lastName)){
+      mLastNameField.setError("Last name cannot have numbers");
+      mLastNameField.requestFocus();
+      valid = false;
+    } else if(containSpecialChar(lastName)) {
+      mLastNameField.setError("Last Name cannot have special characters");
+      mLastNameField.requestFocus();
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:226]: Different cases for name in both sentences

File: T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java [hunk 9]
```diff
@@ -200,6 +200,14 @@ private boolean validateRegistrationForm() {
       mFirstNameField.setError("Required.");
       mFirstNameField.requestFocus();
       valid = false;
+    } else if (containNumber(firstName)){
+      mFirstNameField.setError("First name cannot have numbers");
+      mFirstNameField.requestFocus();
+      valid = false;
+    } else if(containSpecialChar(firstName)) {
+      mFirstNameField.setError("First Name cannot have special characters");
```
Comment (olabusayo) [inline T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java:208]: Different cases for name in both sentences

**Comment by avan989** [comment]
updated

**Pull request merged**